catalog/app/utils/json-schema/json-schema.ts

```diff
@@ -0,0 +1,221 @@
+import Ajv from 'ajv'
+import * as dateFns from 'date-fns'
+import * as R from 'ramda'
+
+type CompoundCondition = 'anyOf' | 'oneOf' | 'not' | 'allOf'
+
+export type JsonSchema = Partial<
+  {
+    $ref: string
+    const: string
+    dateformat: string
+    default: any
+    enum: $TSFixMe[]
+    format: string
+    items: JsonSchema
+    properties: Record<string, JsonSchema>
+    type: string | string[] | JsonSchema[]
+  } & Record<CompoundCondition, JsonSchema[]>
+>
+
+// NOTE: can't use R.prop directly, because R.prop type doesn't allow undefined
+const prop = (
+  property: keyof JsonSchema,
+  optSchema?: JsonSchema,
+): JsonSchema[keyof JsonSchema] => (optSchema ? R.prop(property, optSchema) : undefined)
+
+export const isSchemaArray = (optSchema?: JsonSchema) =>
+  prop('type', optSchema) === 'array'
+
+export const isSchemaObject = (optSchema?: JsonSchema) =>
+  prop('type', optSchema) === 'object'
+
+const isSchemaString = (optSchema?: JsonSchema) => prop('type', optSchema) === 'string'
```

Comment:
actually, seems like this `prop` helper doesnt give us much since we can just write smth like
```js
const isSchemaString = (schema?: JsonSchema = {}) => schema.type === 'string'
```

Reply:
or use optional chaining
```js
const isSchemaString = (schema?: JsonSchema) => schema?.type === 'string'
```